### PR TITLE
Do not include internal error in HTTP response

### PR DESCRIPTION
**Pull request by maditya**
There are similar instances in a couple of other places and I can submit in a separate PR.

### CODE REVIEW

File: endpoints/cookie_sync.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (c *cookieSyncEndpoint) parseRequest(r *http.Request) (usersync.Request, ma
 
 	request := cookieSyncRequest{}
 	if err := jsonutil.UnmarshalValid(body, &request); err != nil {
-		return usersync.Request{}, macros.UserSyncPrivacy{}, fmt.Errorf("JSON parsing failed: %s", err.Error())
+		return usersync.Request{}, macros.UserSyncPrivacy{}, fmt.Errorf("request JSON is invalid")
```
Comment (SyntaxNode) [inline endpoints/cookie_sync.go:125]: While I agree it's better not to directly return the error message, a generalized error does not provide any information about what was invalid. Thoughts?

Comment (maditya) [reply endpoints/cookie_sync.go:125]: Would "Malformed JSON" be more intuitive? The user should be able to validate the json using one of the online or command-line verifiers, which can provide more specific instructions. Would that suffice?

Comment (SyntaxNode) [reply endpoints/cookie_sync.go:125]: My concern is the request json is ephemeral and may not be available, whereas the error is more likely to be logged. 

Let's take a step back, what issue are you targeting to fix with this change?

Comment (maditya) [reply endpoints/cookie_sync.go:125]: > what issue are you targeting to fix with this change?

To not expose implementation specific errors to the client, as a security best practice. Logging the specific error message and/or capturing metrics around it seems more reasonable.

**Comment by SyntaxNode** [comment]
> To not expose implementation specific errors to the client, as a security best practice.

I hear ya. I'm not personally too concerned with the security aspect since this is an open source project, the stack trace shouldn't reveal anything not already included here (except for custom changes).

> Logging the specific error message and/or capturing metrics around it seems more reasonable.

Pretty sure we already have metrics. We don't log request specific errors since Prebid Server traffic volume is too high. That's a job for the publisher's or host's analytics systesm.

I'm hesitant to accept this right now since it's a breaking change. We started a conversation about introducing breaking changes in PBS 3.0? for responses in issue $3556. How about we move this discussion there and close this PR for now. You're welcomed to contribute to the decisions made in that thread.

**Comment by SyntaxNode** [comment]
Perfect. Thank you for your understanding.

**Pull request closed without merging**